services/QuillLMS/app/serializers/diagnostic_insights/student_learning_sequence_activity_serializer.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module DiagnosticInsights
+  class StudentLearningSequenceActivitySerializer < ActiveModel::Serializer
+    # Defined above constants since we use this method to build some of them
+    def self.recommendations(diagnostic_ids)
+      UnitActivity.joins(unit: { unit_template: :recommendations })
+        .where(unit: { unit_template: { recommendations: { activity_id: diagnostic_ids } } })
+        .pluck(:activity_id)
+    end
+
+    PRE_DIAGNOSTIC_IDS = Activity::ALL_PRE_TEST_DIAGNOSTIC_IDS
+    POST_DIAGNOSTIC_IDS = Activity.where(id: PRE_DIAGNOSTIC_IDS).pluck(:follow_up_activity_id)
+    PRE_RECOMMENDATION_IDS = recommendations(PRE_DIAGNOSTIC_IDS)
+    POST_RECOMMENDATION_IDS = recommendations(POST_DIAGNOSTIC_IDS)
+
+    PRE_DIAGNOSTIC = 'pre_diagnostic'
+    POST_DIAGNOSTIC = 'post_diagnostic'
+    PRE_RECOMMENDATION = 'pre_recommendation'
+    POST_RECOMMENDATION = 'post_recommendation'
+
+    attributes :activity_id, :activity_type, :completed_at, :created_at, :activity_session_id, :proficient
+
+    def proficient
+      return nil unless pre_diagnostic? || post_diagnostic?
+
+      object.activity_session_unscoped&.concept_results&.all?(&:correct)
+    end
```

Comment:
Just confirming that we want to define proficiency as having all concept results correct? That seems like a high bar.

Reply:
Good question.  Peter S flagged this during the spec-writing process, but we're deferring to Curriculum on this one.  We may change our minds later about what counts as "proficient", but right now it's getting every question correct.

Reply:
@anathomical but is having all concept results correct the same as having every question correct? Is it not possible to have like an incorrect concept result for spelling but still get the question's concept and therefore the question itself right?